src/Text/Layout/Table/Cell.hs

```diff
@@ -29,6 +29,34 @@ class Cell a where
     dropBoth :: Int -> Int -> a -> a
     dropBoth l r = dropRight r . dropLeft l
 
+    -- | Like 'dropLeft', but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of padding needed is returned with the
+    -- result.
+    dropLeftNoPad :: Int -> a -> (Int, a)
+    dropLeftNoPad n a = (0, dropLeft n a)
+
+    -- | Like 'dropRight, but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of padding needed is returned with the
+    -- result.
+    dropRightNoPad :: Int -> a -> (a, Int)
+    dropRightNoPad n a = (dropRight n a, 0)
+
+    -- | Like 'dropBoth, but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of left and right padding needed is
+    -- returned with the result.
+    -- The default implementation, when invoked with @dropBothNoPad l r@, will
+    -- first drop @l@ from the left and then will drop an amount from the right
+    -- so that the total dropped is @l + r@. Note that this means it may drop
+    -- less than @r@ from the right if it is forced to drop more than @l@ from
+    -- the left.
```

Comment:
I have a hunch about what this function does.  But from the documentation alone I cannot understand the purpose or context.

* When does `dropBoth` add padding? Is this in the context of multi-width characters?
* When or why would someone use the function?
* Could you more deterministically describe the result? :)

Reply:
I think I've improved the documentation, but let me know if you feel it's still lacking.

Also note that all these comments actually refer to commits in #37. I've responded to them here, but the changes will show up in that PR.